sdk/search/azure-search/src/main/java/com/azure/search/Document.java

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.search;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Represents a document
+ * <p>
+ * A property bag is used for scenarios where the index schema is only known at run-time
+ * <p>
+ * If the schema is known, user can convert the properties to a specific object type
+ */
+public class Document extends HashMap<String, Object> {
```

Comment:
@brjohnstmsft should we have an 'id' or 'key' property here by default?

Reply:
No, because the name of the key field can change depending on the customer's index definition. This class is equivalent to the `Document` class in the .NET SDK, as expected.